_inc/client/state/initial-state/actions.js

```diff
@@ -3,11 +3,7 @@
  */
 import { JETPACK_SET_INITIAL_STATE } from 'state/action-types';
 
-export const setInitialState = () => {
-	return dispatch => {
-		dispatch( {
-			type: JETPACK_SET_INITIAL_STATE,
-			initialState: window.Initial_State,
-		} );
-	};
-};
+export const setInitialState = () => ( {
+	type: JETPACK_SET_INITIAL_STATE,
+	initialState: window.Initial_State,
```

Comment:
We should probably parametrize the action for `initialState` to put the burden of instantiating it with `window.Initial_State` on the consumer.